### PR TITLE
Add more pratical descriptions to App CRs

### PR DESCRIPTION
**Pull request by pipo02mix**
## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by axbarsan — APPROVED** [review]

File: docs/crd/application.giantswarm.io_app.yaml [hunk 1]
```diff
@@ -16,47 +16,49 @@ spec:
...
               description: |
-                Configuration details for the app.
+                Cluster configuration details for the application injected by cluster-operator with basic
+                information of the cluster where the App is deployed. 
               properties:
                 configMap:
                   description: |
-                    If present, points to a ConfigMap resource that holds configuration data
-                    used by the app.
+                    Defines the reference of a "ConfigMap" where is saved the cluster configuration values
```
Comment (rossf7) [inline docs/crd/application.giantswarm.io_app.yaml:35]:
```suggestion
                    Defines a reference to a "ConfigMap" where is saved the cluster configuration values 
```

File: docs/crd/application.giantswarm.io_app.yaml [hunk 2]
```diff
@@ -78,12 +80,15 @@ spec:
...
-                    properties must not be set.
+                    Defines whether to use inCluster credentials. If true, it uses the service account to 
+                    authenticate against the Kubernetes API in the same cluster where the app-operator
+                    is running. In that case, the context and secret properties must not be set. If false,
+                    secret and context is used by the app-operator to access the external cluster API.
                   type: boolean
                 secret:
                   description: |
-                    References a Secret resource holding the kubeconfig details, if not using inCluster credentials.
+                    Defines the reference of a "Secret" where is saved the kubeconfig configuration
```
Comment (rossf7) [inline docs/crd/application.giantswarm.io_app.yaml:90]:
```suggestion
                    Defines the reference to a "Secret" where is saved the kubeconfig configuration
```

File: docs/crd/application.giantswarm.io_app.yaml [hunk 3]
```diff
@@ -104,15 +109,18 @@ spec:
...
               description: |
-                Kubernetes namespace in which to install the workloads defined by this App.
+                Kubernetes namespace in which to install the workloads defined by this App in the 
+                given cluster.
               type: string
             userConfig:
               description: |
-                Additional and optional user-provided configuration for the app.
+                User configuration for the App. This configuration will be merged with cluster configuration
+                one to generate a single "ConfigMap" in the target cluster.
```
Comment (rossf7) [inline docs/crd/application.giantswarm.io_app.yaml:118]:
```suggestion
                User configuration for the App. This configuration will be merged with the catalog and
                cluster configuration to generate a single values "ConfigMap" in the target cluster.
```

File: docs/crd/application.giantswarm.io_appcatalog.yaml [hunk 4]
```diff
@@ -15,18 +15,17 @@ spec:
   validation:
     openAPIV3Schema:
       description: |
-        An AppCatalog defines a collection of apps to install in one or several clusters.
+        Defines a location where packaged applications are stored and shared,
+        ready to be install in a Kubernetes cluster.
```
Comment (rossf7) [inline docs/crd/application.giantswarm.io_appcatalog.yaml:19]:
```suggestion
        ready to be installed in a Kubernetes cluster.
```

File: docs/crd/application.giantswarm.io_appcatalog.yaml [hunk 5]
```diff
@@ -77,8 +76,9 @@ spec:
                   type: string
                 type:
                   description: |
-                    Indicates which repository type would be used for this AppCatalog.
-                    Example: 'helm'
+                    Determines the type of storage. Currently only `helm` is available.
+                    - Helm type storage use the known Helm Chart Repository format to store the chart
+                      packages and expose some metadata file `index.yaml` to manage the catalog.
```
Comment (rossf7) [inline docs/crd/application.giantswarm.io_appcatalog.yaml:81]:
```suggestion
                      packages and expose some metadata in the file `index.yaml` to manage the catalog.
```

**Review by rossf7 — APPROVED** [review]
Nice, LGTM

Just a few minor things.

File: docs/crd/application.giantswarm.io_app.yaml (hunk 1 above)
Comment (marians) [reply docs/crd/application.giantswarm.io_app.yaml:35]: No need to wrap the word ConfigMap in quotes

File: docs/crd/application.giantswarm.io_app.yaml (hunk 2 above)
Comment (marians) [reply docs/crd/application.giantswarm.io_app.yaml:90]: Same here with Secret (no quotes)

**Review by tomahawk28 — APPROVED** [review]
LGTM :)

**Comment by marians** [comment]
You have to run the unit test with `-u` in order to update the CR and CRD files, then commit both files to your branch.

**Comment by pipo02mix** [comment]
@marians done

**Pull request merged**